### PR TITLE
Handle named embedded structs, doc comments; fix optional types in TS

### PR DESCRIPTION
**Pull request by kyegupov**
* properly handle embedded structs with json name tag
* retain doc comments for fields
* fix optional type (?) in Typescript (the previous implementation was broken)
* ensured the tests are passing

### CODE REVIEW

File: examples/example.go [hunk 1]
```diff
@@ -40,10 +40,10 @@ type Nested struct {
 	Person struct{} `json:"person"`
 }
 
-// Embedded will take all types from the embedded types and insert them in to the new type.
-type Embedded struct {
-	Person
-}
+// // Embedded will take all types from the embedded types and insert them in to the new type.
```
Comment (natdm) [inline examples/example.go]: Can we delete this if we don't need it?

File: template/fragments.go [hunk 2]
```diff
@@ -49,17 +50,20 @@ var elmTemplates = langTemplates{
 	arrayShortOpen:  ` List`,
 	arrayShortClose: ``,
 	basic:           ` {{updateElmType .Type}}`,
-	comment:         ` {{elmComment .Comment}}`,
+	fieldDocComment: `{{elmMultilineComment .DocComment 1}}`,
```
Comment (natdm) [inline template/fragments.go:53]: Should this have a space at the start like the previous `comment` field had to keep it consistent?

**Review by natdm — APPROVED** [review]
Super small changes. Great work (again). They're small enough that I approved regardless.

File: examples/example.go (hunk 1 above)
Comment (kyegupov) [reply examples/example.go]: Good point. Done.

File: template/fragments.go (hunk 2 above)
Comment (kyegupov) [reply template/fragments.go:53]: First, this is a multiline comment that precedes the field, so it does not need a space.

Second, for the single-line comment (handled in the `fieldClose` fragment), the whitespace is now prepended by the `func lineComment` that implements it.

**Comment by kyegupov** [comment]
@natdm thank you! But, regardless of approval, only you can merge this :)

**Comment by natdm** [comment]
@kyegupov good work!

**Pull request merged**